### ISSUE TITLE
Status for new release - no new 2.x point release since December 2021.

### ISSUE THREAD

**Issue by MrPippin66** [NagiosEnterprises/ncpa#907]
This isn't a bug report, but the last release was over a year ago.

Do we have a tentative schedule for the next release?

**Comment by ericloyd** [comment]
https://github.com/NagiosEnterprises/ncpa/milestone/8 would be a good place to see where things stand.

**Comment by MrPippin66** [comment]
I understand, but the only milestone is for 3.0, which is a major release. I'm referring to a point release for 2.x branch.

**Comment by ericloyd** [comment]
Sorry then.  I can't help you.  @jomann09, and clues?

**Comment by jomann09** [comment]
I'm sorry @ericloyd I am no longer a maintainer on this project, so I am not sure what the status is currently.

**Comment by ericloyd** [comment]
Whoops!  Sorry Jake!

**Comment by ericloyd** [comment]
@ccztux is it you? Bueller?  Anyone?  Who's maintaining this now?

**Comment by jomann09** [comment]
@sawolf may be able to give a response

**Comment by ericloyd** [comment]
I look forward to finding out.

**Comment by ccztux** [comment]
> @ccztux is it you? Bueller? Anyone? Who's maintaining this now?

No, I am only a community contributor.

**Comment by ccztux** [comment]
> I'm sorry @ericloyd I am no longer a maintainer on this project, so I am not sure what the status is currently.

Thats so sad you did a great job.

**Comment by bramassendorp** [comment]
This all worries me for a enterprise grade product, you should expect some maintenance etc.

Also for the 3.x version, I have seen no news lately.

Is NCPA dead?

**Comment by MrPippin66** [comment]
To add to that, OpenSSL libraries are also part of the package, which also need to be updated.

My primary concern (outside security) is there are several issues in the current version of PSUTIL bundled in NCPA which are fixed in the current codebase for that, and are currently unresolved in the shipping NCPA.

**Comment by ericloyd** [comment]
Considering Nagios Enterprises pushes NCPA as the current, official, preferred agent for commercial enterprise Nagios deployments, it's getting to the "unacceptable" stage that there is apparently no maintenance of this project at this point.

I will be bringing this up with Nagios Enterprises immediately.

**Comment by jomann09** [comment]
Just as a side note @pittagurneyi, the NCPA project is generally an "open source" project but it is managed my Nagios Enterprises. I have since left the company which is why I am no longer maintaining it, and the work on it was generally limited from the company side. I never did get the time to fully build out the dev-v3 version and make the required changes.

I have had an idea for an agent that does not use python that would resolve a lot of the issues that NCPA has had over the years, while keeping a similar structure. The other big issue is that NCPA is not using a normal open source license, which I think also hurts the ability for community help... along with the complexity of the build process.

**Comment by ericloyd** [comment]
My company is Nagios Authorized Partner and we've initiated communications with our partner contacts at Nagios to see what the status of NCPA is.  Unfortunately, it seems to be a common thing with Nagios Enterprises that some piece of code gets created and developed for a while, and then left alone when it starts to get hard to maintain, update, or secure.  I really hope that won't be the case here, since my company has done a lot of business selling people on using NCPA as the Nagios agent, which is what Nagios Enterprises has been pushing since October of 2013.

**Comment by sawolf** [comment]
Hi Everyone - sorry for the delay in response. We're currently looking at another release for 2.4.x in early February. We definitely want to get NCPA 3 out relatively soon, for all the same reasons that were mentioned above. Currently, we're currently ramping up another developer on this project, and they're working on some issues in 2.4.x at the moment.

**Comment by ericloyd** [comment]
> Hi Everyone - sorry for the delay in response. We're currently looking at another release for 2.4.x in early February. We definitely want to get NCPA 3 out relatively soon, for all the same reasons that were mentioned above. Currently, we're currently ramping up another developer on this project, and they're working on some issues in 2.4.x at the moment.

Thanks, Sebastian.  Maybe start putting those release dates in the timeline so people don't freak out like we are now?  :-)

**Comment by MrPippin66** [comment]
I'll posit the same. Thanks.

In the same vein, really need to try and establish a release timeline for V3, because at this point, there has been no new releases for AIX in almost 3 years, and all pending V3 due to the requirement to switch to python3 to resolve known build issues.

**Comment by ccztux** [comment]
> Just as a side note @pittagurneyi, the NCPA project is generally an "open source" project but it is managed by Nagios Enterprises. I have since left the company which is why I am no longer maintaining it, and the work on it was generally limited from the company side. I never did get the time to fully build out the dev-v3 version and make the required changes.
> 
> I have had an idea for an agent that does not use python that would resolve a lot of the issues that NCPA has had over the years, while keeping a similar structure. The other big issue is that NCPA is not using a normal open source license, which I think also hurts the ability for community help... along with the complexity of the build process.

I am eager to see if you will start your own monitoring agent project.

**Comment by ericloyd** [comment]
@pittagurneyi is correct that NCPA cannot be described as open source with the limitations of the license imposed by Nagios Enterprises.  Especially since that license is the generic license included with commercial Nagios software.  Combined with https://github.com/NagiosEnterprises/ncpa/blob/master/LICENSE.rst, which states:

> 3.  The Software may not be Forked
> 5.  Any Contribution of materials to the Software by any User is subject to, and signifies, either explicitly or implicitly, the User's acceptance of the terms and conditions of the Nagios Contributor Agreement
>  (which in turn states, "you hereby assign to us joint ownership")
> 6.   Any standalone use of the Software, or use of the Software in conjunction with products, projects, or other software not authored or distributed by the Company - including third-party software that competes with the offerings of Company - is strictly prohibited and is a direct violation of the terms of the license.

There are other examples.  NCPA is NOT open source software.  https://en.wikipedia.org/wiki/Open_source states that open source is:

> Generally, open source refers to a computer program in which the source code is available to the general public for use or modification from its original design. Code is released under the terms of a software license. Depending on the license terms, others may then download, modify, and publish their version (fork) back to the community.

The problem is that Nagios Enterprises will never address this or change it because there is no incentive to.  Most of the users of Nagios XI who use NCPA do so by using a wizard which installs it for them, configures it for them, and they get their monitoring and pay their license fee for Nagios XI.  NRPE isn't even a Nagios supported product (and it never truly was) because the number 1 Windows-based NRPE client (and continues to be) a third-party developed program called NSClient++ which may or may not be discontinued (depending on where you look).

This has never really bothered me in the past because I assumed that Nagios Enterprises was maintaining NCPA as it has been promoting it for the past decade as the new and improved agent for all modern operating systems.  We may need to rethink that strategy and I will be bringing this up with our CEO and that of Nagios Enterprises for a conversation.

**Comment by ericloyd** [comment]
Nothing positive to share.

**Comment by ericloyd** [comment]
I was told by Nagios Enterprise management that "Cory and Sebastian are on it.  I just checked and they will update the Github thread with updates."

That was January 25th.  Other than @sawolf's comment later that same day, there has been no further communication either privately or here on this thread.

**Comment by corynorell** [comment]
Hello everyone,

I will be managing the project from hereon out; that means ensuring roadmaps/timelines are kept updated with needed changes and - more importantly - adhered to as best we can. To that end, I will be allocating additional development resources to the project. Please keep in mind that it will take some time to onboard additional developers to the project.

NCPA 2.4.1 is currently in the final testing stages and should be released in the next week or two. After that - pending scoping the work - we will either be updating NCPA 2 to use Python 3, or working on taking NCPA 3 to completion.

**Comment by MrPippin66** [comment]
Thanks! When can we expect the milestones to be updated?

**Comment by corynorell** [comment]
I'm going to try to have the milestones updated this upcoming week, by Friday 2/17 at the latest.

**Comment by bramassendorp** [comment]
So basically,

When can we expect the 2.4.1 version? And when can we expect the promised improved version 3.0 with all the new features?

**Comment by corynorell** [comment]
As stated above, NCPA 2.4.1 was in the final testing stages prior to release. It has now been released.

As for 3.0.0 and beyond, I updated the milestones as promised. NCPA 3 will be released as soon as it's stable with the goal of providing a Python 3 based agent. Other feature requests and fixes will be regularly released in incremental iterations thereafter. 

Closing this thread as I have already made our intentions clear.
